### ISSUE TITLE
Installation failure

### ISSUE THREAD

**Issue by wisdom-17** [FriendsOfSymfony/FOSJsRoutingBundle#251]
Hi all,

I am trying to install the FOSJsRoutingBundle using the command: "`composer require friendsofsymfony/jsrouting-bundle`"

This results in the following error:
"Problem 1
    - The requested package friendsofsymfony/user-bundle (locked at 2.0.x-dev) is satisfiable by friendsofsymfony/user-bundle[2.0.x-dev] but these conflict with your requirements or minimum-stability.

Installation failed, reverting ./composer.json to its original content."

Below is my composer.json file. Appreciate any advice.

```
{
    "name": "root/periscope.project",
    "license": "proprietary",
    "type": "project",
    "autoload": {
        "psr-4": {
            "": "src/"
        },
        "classmap": [
            "app/AppKernel.php",
            "app/AppCache.php"
        ]
    },
    "autoload-dev": {
        "psr-4": {
            "Tests\\": "tests/"
        }
    },
    "require": {
        "php": ">=5.5.9",
        "symfony/symfony": "3.0.*",
        "doctrine/orm": "^2.5",
        "doctrine/doctrine-bundle": "^1.6",
        "doctrine/doctrine-cache-bundle": "^1.2",
        "symfony/swiftmailer-bundle": "^2.3",
        "symfony/monolog-bundle": "^2.8",
        "sensio/distribution-bundle": "^5.0",
        "sensio/framework-extra-bundle": "^3.0.2",
        "incenteev/composer-parameter-handler": "^2.0",
        "symfony/assetic-bundle": "^2.7"
    },
    "require-dev": {
        "sensio/generator-bundle": "^3.0",
        "symfony/phpunit-bridge": "^2.7",
        "doctrine/doctrine-fixtures-bundle": "^2.3"
    },
    "scripts": {
        "post-install-cmd": [
            "Incenteev\\ParameterHandler\\ScriptHandler::buildParameters",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::buildBootstrap",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::clearCache",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installAssets",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installRequirementsFile",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::prepareDeploymentTarget"
        ],
        "post-update-cmd": [
            "Incenteev\\ParameterHandler\\ScriptHandler::buildParameters",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::buildBootstrap",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::clearCache",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installAssets",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installRequirementsFile",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::prepareDeploymentTarget"
        ]
    },
    "config": {
        "platform": {
            "php": "5.5.9"
        }
    },
    "extra": {
        "symfony-app-dir": "app",
        "symfony-bin-dir": "bin",
        "symfony-var-dir": "var",
        "symfony-web-dir": "web",
        "symfony-tests-dir": "tests",
        "symfony-assets-install": "relative",
        "incenteev-parameters": {
            "file": "app/config/parameters.yml"
        }
    }
}
```

**Comment by xabbuh** [comment]
This looks weird. The command is requiring the FOSJsRoutingBundle while the error message mentions the FOSUserBundle which is not required by the former one. Are you sure that you didn't make any typo in the `require` command?

**Comment by symio** [comment]
Similar error unable to upgrade sf 2.8 to 3.3 jsrouting stay locked @1.5.4 and i'm unable to make composer migrate it to 2.*@alpha or 2.*@dev with the "but these conflict with your requirements or minimum-stability" alert.

This is very bad because 80% of the app logic is based upon ajax calls with jsrouting interactions and sf 2.8 is getting out of life in a month.

Please help ! (the fifth element)
**solved :**
Edit : While i was writing theses lines i had a weird idea and it worked :

For those who need to upgrade from 2.8 to 3.* :
  - update jsrouting to 2.*@dev (put this in composer.json : "friendsofsymfony/jsrouting-bundle": "2.*@dev",) then perform $ composer update friendsofsymfony/jsrouting-bundle by letting symfony/symfony to 2.8.*
  - u are now able to migrate sf to 3.* following the migration guide : http://symfony.com/doc/current/setup/upgrade_major.html#upgrade-major-symfony-composer

**Comment by tobias-93** [comment]
Hi @symio,

I'm sorry for your troubles, this problem is caused by the fact that the latest stable version 1.x doesn't support SF 3.x and higher, and we did not release a stable version in the 2.x branch yet.
However, according to http://symfony.com/roadmap?version=2.8, SF 2.8 will be supported with bugfixes until November 2018, so it will not be a problem in the near future for the more conservative users.
I think we could tag a stable release for 2.x, but I would have to have a look into the changes and write some upgrade documentation first.

**Comment by symio** [comment]
Thanks @tobias-93 for the answer,

This run was a test on an internal appliance that could stay in 2.8.* : so no big deal here
I use your bundle in a customer application and it is a bit more problematic as we have a contract that we will have to migrate the application to 3.4 as soon as the symfony version will be released in a month.
I will conduct compatibility tests with your bundle's version (in dev 2.*) to find the more "stable" one prior to upgrade ; I could keep you in touch if you like.

**Comment by tobias-93** [comment]
@symio You're welcome.
Of course I'm happy to hear all the issues you run into!
I think I can manage to get the release to 2.0 out within a month, but of course I would advise you to use the development version already in the mean time to speed up your migration process.

**Comment by tobias-93** [comment]
I have released version 2.0.0, please let me know if you run into any issues.
